lib/kennel/models/record.rb

```diff
@@ -129,25 +135,53 @@ def build_json
         }.compact
       end
 
+      # One of three outcomes:
+      # - throws (on non-validation error)
+      # - returns, validation_errors is not empty, and as_json is nil (invalid)
+      # - returns, validation_errors is empty, and as_json is not nil (valid)
+      def prepare
```

Comment:
less generic name would be nice
prepare_validation ?

Reply:
One of the things I'd love to do at some point would be to split the model class instance methods into two: pre-prepare and post-prepare. 

- pre-prepare has settings, and knows how to build_json / validate_json / prepare
- post-prepare has things like add_tracking_id, diff, resolve.

So conceptually calling the code ends up looking like:

```
monitor = K::M::Monitor.new(....) { ... }
built = monitor.build
built.as_json
built.add_tracking_id
# etc 
```

At which point "prepare" is basically the same as / replaced by "build".

For now, we could still call it `build` or `build_and_validate` if you like.

Reply:
I've called it `build`.